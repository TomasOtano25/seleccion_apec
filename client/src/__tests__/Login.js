import React from "react";
import { create } from "react-test-renderer";
import { render, cleanup, fireEvent } from "react-testing-library";
import { MemoryRouter } from "react-router-dom";
import "jest-dom/extend-expect";
import LoginPage from "../pages/Login";

afterEach(cleanup);

describe("Login page", () => {
  it("should matches the snapshot", () => {
    const loginPage = create(
      <MemoryRouter>
        <LoginPage />
      </MemoryRouter>
    );

    expect(loginPage.toJSON()).toMatchSnapshot();
  });

  it("should change the entry email change and the button be disabled", () => {
    const { getByPlaceholderText, getByTestId } = render(
      <MemoryRouter>
        <LoginPage />
      </MemoryRouter>
    );

    const emailInput = getByPlaceholderText("Email");
    fireEvent.change(emailInput, {
      target: { value: "20180089@unapec.edu.do" }
    });

    const submitButton = getByTestId("submit-button");

    expect(emailInput.value).toBe("20180089@unapec.edu.do");

    expect(submitButton).not.toBeDisabled();
  });
  it("should submit", () => {});
  it("should not submit if required fields are empty", () => {});
});
