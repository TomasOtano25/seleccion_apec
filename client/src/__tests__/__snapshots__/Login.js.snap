// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page it matches the snapshot 1`] = `
<div
  className="limiter"
>
  <div
    className="container-login100"
  >
    <div
      className="wrap-login100"
    >
      <div
        className="mb-4"
      >
        <img
          height="auto"
          src="https://cdn.unapec.edu.do/portal-dynamic/Documentos/descargas/logo_principal/logo-unapec_color.png"
          width={275}
        />
      </div>
      <form
        className="login100-form validate-form"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <span
          className="login100-form-title"
        >
          Iniciar sesión
        </span>
        <div
          className="wrap-input100 validate-input "
        >
          <input
            className="input100"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          <span
            className="focus-input100"
          />
          <span
            className="symbol-input100"
          >
            <i
              aria-hidden="true"
              className="fas fa-envelope"
            />
          </span>
        </div>
        <div
          className="container-login100-form-btn"
        >
          <button
            className="login100-form-btn"
            data-testid="submit-button"
            disabled={false}
            type="submit"
          >
            <span>
              Login
            </span>
          </button>
        </div>
      </form>
      <div
        className="mt-2"
      >
        <a
          href="/admission"
          onClick={[Function]}
        >
          Registrar
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Login page should matches the snapshot 1`] = `
<div
  className="limiter"
>
  <div
    className="container-login100"
  >
    <div
      className="wrap-login100"
    >
      <div
        className="mb-4"
      >
        <img
          height="auto"
          src="https://cdn.unapec.edu.do/portal-dynamic/Documentos/descargas/logo_principal/logo-unapec_color.png"
          width={275}
        />
      </div>
      <form
        className="login100-form validate-form"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <span
          className="login100-form-title"
        >
          Iniciar sesión
        </span>
        <div
          className="wrap-input100 validate-input "
        >
          <input
            className="input100"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          <span
            className="focus-input100"
          />
          <span
            className="symbol-input100"
          >
            <i
              aria-hidden="true"
              className="fas fa-envelope"
            />
          </span>
        </div>
        <div
          className="container-login100-form-btn"
        >
          <button
            className="login100-form-btn"
            data-testid="submit-button"
            disabled={false}
            type="submit"
          >
            <span>
              Login
            </span>
          </button>
        </div>
      </form>
      <div
        className="mt-2"
      >
        <a
          href="/admission"
          onClick={[Function]}
        >
          Registrar
        </a>
      </div>
    </div>
  </div>
</div>
`;
